### ISSUE TITLE
Returning UIImage in custom ImageSourceService doesn't trigger an UI update

### ISSUE THREAD

**Issue by roubachof** [dotnet/maui#13490]
### Description

Hi there !

https://github.com/roubachof/Maui.Nuke has a weird issue when returning a valid UIImage with ImageSourceServiceResult.
It doesn't seem to trigger the display of the image right away...
But if you force the layout to refresh, the UIImage renders correctly.

### Steps to Reproduce

1. Create new MAUI app with the default template
2. Clone https://github.com/roubachof/Maui.Nuke next to your app
3. Reference Maui.Nuke csproj in your app
4. Initialize Nuke as described in Maui.Nuke 
5. put a breakpoint in https://github.com/roubachof/Maui.Nuke/blob/d8682d833abd7cd6bafb2e82e46e5e72d3823b47/Maui.Nuke/NukeFileImageSourceService.cs#L105
6. You will see UIImage is not null and valid, but you won't see the bot displayed
7. If you click the button, image appears

### Link to public reproduction project repository

https://github.com/roubachof/Maui.Nuke

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS

### Did you find any workaround?

nope

### Relevant log output

```shell
nothing here
```

**Comment by vhugogarcia** [comment]
I'm facing this issue as well on iOS. 

@PureWeen do you know if this can be fixed by using a custom handler in .NET MAUI somehow? 🤔 

Thanks in advance

**Comment by vhugogarcia** [comment]
Maybe this can improve performance a lot for Maui in iOS when rendering images lots os images in a collection view or bindable layout. @jonathanpeppers what do you think? Maybe integrate it natively on MAUI? 

Thank you

**Comment by homeyf** [comment]
Verified this issue with VSM 17.6.7 (build 417). Can repro on ios platform.

**Comment by roubachof** [comment]
Just tested with .net8 MAUI, and still failing

**Comment by roubachof** [comment]
I tested it with .net 7 and .net 8, same issue

**Comment by roubachof** [comment]
No. Nearly 2 years old issue preventing a fast caching native lib to work....
I'm not very happy about this.

**Comment by roubachof** [comment]
Boooooh I am the ghost of Sharpnado and I'm here to torment your souls

@PureWeen @jsuarezruiz

**Comment by roubachof** [comment]
You can find a new repro sample here: https://github.com/roubachof/Maui.Nuke/issues/9

**Comment by roubachof** [comment]
@PureWeen I found the issue in the maui code: https://github.com/dotnet/maui/blob/971e326e7cf59c186b26fb1e02c3e2e46c882b19/src/Core/src/Handlers/Image/ImageHandler.iOS.cs#L70

Since Nuke is File or URI image source, it's not invalidated and then sometimes the image doesn't show.
I created a custom image handler in nuke latest version 12.8.2, and now it works as intended
